docs/source/examples/dcgan.rst

```diff
@@ -0,0 +1,358 @@
+And the distribution :math:`p_{\mathrm model}({\bf s})`
+generated by :math:`G({\bf x})` can match the true distribution :math:`p({\bf s})`.
+
+.. math::
+    \min_{G} \max_{D} \mathbb{E}_{p(x)} \log D(x) + \mathbb{E}_{p_{\mathrm{model}}} \log (1-D(x))
+
+When training actually, the above min-max problem is solved by alternately updating the
+discriminator :math:`D({\bf s})` and the generator :math:`G({\bf x})` [4].
+
+.. figure:: ../../image/dcgan/update-gan.png
+   :scale: 100%
+
+   cited from [4]
+
+1.3 What is DCGAN?
+-------------------
+
+In this section, we will introduce the model called DCGAN(Deep Convolutional GAN) proposed by Radford et al.[5].
+As shown below, it is a model using CNN(Convolutional Neural Network) as its name suggests.
+
+.. figure:: ../../image/dcgan/dcgan.png
+   :scale: 100%
+
+   cited from [5]
+
+In addition, although GAN is known for its difficulty in learning, this paper introduces various techniques
+for successful learning:
+
+1. Convert max-pooling layers to convolution layers
+2. Convert fully connected layers to global average pooling layers in the discriminator
+3. Use batch normalization layers in the generator and the discriminator
+4. Use leaky ReLU activation functions in the discriminator
+
+2. Implementation of DCGAN in Chainer
+=======================================
+
+There is an example of DCGAN in the official repository of Chainer,
+so we will explain how to implement DCGAN based on this:
+`chainer/examples/dcgan <https://github.com/chainer/chainer/tree/master/examples/dcgan>`_
+
+2.1 Define the generator model
+-------------------------------
+
+First, let's define a network for the generator.
+
+.. literalinclude:: ../../../examples/dcgan/net.py
+   :language: python
+   :pyobject: Generator
+   :caption: train_dcgan.py
+
+When we make a network in Chainer, we should follow some rules:
+
+1. Define a network class which inherits :class:`~chainer.Chain`.
+2. Make :class:`chainer.links` 's instances in the ``init_scope():`` 
+   of the initializer ``__init__``.
+3. Concatenate :class:`chainer.links` 's instances with :class:`chainer.functions`
+   to make the whole network.
+
+If you are not familiar with constructing a new network, you can read
+:ref:`this tutorial<creating_models>`.
+
+As we can see from the initializer ``__init__``, the ``Generator``
+uses the deconvolution layer :class:`~chainer.links.Deconvolution2D`
+and the batch normalization :class:`~chainer.links.BatchNormalization`.
+In ``__call__``, each layer is concatenated by :class:`~chainer.functions.relu`
+except the last layer.
+
+Because the first argument of ``L.Deconvolution`` is the channel size of input and
+the second is the channel size of output, we can find that each layer halve the
+channel size. When we construct ``Generator`` with ``ch=1024``, the network
+is same with the image above.
+
+.. note::
+    Be careful when you concatenate a fully connected layer's output and
```

Comment:
Fixed as
> Be careful when passing the output of a fully connected layer to a convolution layer, because the convolutional layer needs additional dimensions for inputs. As we can see the 1st line of __call__, the output of the fully connected layer is reshaped by reshape to add the dimensions of the channel, the width and the height of images.